huaweicloud/services/rms/resource_huaweicloud_rms_remediation_execution.go

```diff
@@ -0,0 +1,287 @@
+							Description: `The cloud service name.`,
+						},
+						"resource_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The resource type.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceRemediationExecutionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+
+	var (
+		createRemediationExecutionHttpUrl = "v1/resource-manager/domains/{domain_id}/policy-assignments/{policy_assignment_id}/remediation-execution"
+		createRemediationExecutionProduct = "rms"
+	)
+	client, err := conf.NewServiceClient(createRemediationExecutionProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RMS client: %s", err)
+	}
+
+	policyAssignmentId := d.Get("policy_assignment_id").(string)
+	createRemediationExecutionPath := client.Endpoint + createRemediationExecutionHttpUrl
+	createRemediationExecutionPath = strings.ReplaceAll(createRemediationExecutionPath, "{domain_id}", conf.DomainID)
+	createRemediationExecutionPath = strings.ReplaceAll(createRemediationExecutionPath, "{policy_assignment_id}", policyAssignmentId)
+
+	createRemediationExecutionOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	createRemediationExecutionOpt.JSONBody = utils.RemoveNil(buildCreateRemediationExecutionBodyParams(d))
+	_, err = client.Request("POST", createRemediationExecutionPath, &createRemediationExecutionOpt)
+	if err != nil {
+		return diag.Errorf("error creating RMS remediation execution: %s", err)
+	}
+
+	d.SetId(policyAssignmentId)
+
+	err = waitingForRemediationExecutionCompleted(ctx, client, d, conf.DomainID)
+	if err != nil {
+		return diag.Errorf("error waiting for RMS remediation execution completed: %s", err)
+	}
+
+	return resourceRemediationExecutionRead(ctx, d, meta)
+}
+
+func buildCreateRemediationExecutionBodyParams(d *schema.ResourceData) map[string]interface{} {
+	resourceIds := d.Get("resource_ids").(*schema.Set).List()
+	bodyParams := map[string]interface{}{
+		"all_supported": d.Get("all_supported"),
+		"resource_ids":  utils.ValueIgnoreEmpty(resourceIds),
+	}
+	return bodyParams
+}
+
+func waitingForRemediationExecutionCompleted(ctx context.Context, client *golangsdk.ServiceClient, d *schema.ResourceData, domainId string) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			policyAssignmentId := d.Get("policy_assignment_id").(string)
+			checkExpression := "[?state=='IN_QUEUE' || state=='IN_PROGRESS']"
+			result, err := getRemediationExecution(client, domainId, policyAssignmentId)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			pendingTasks := utils.PathSearch(checkExpression, result, make([]interface{}, 0)).([]interface{})
+			if len(pendingTasks) == 0 {
+				return result, "COMPLETED", nil
+			}
+
+			return nil, "PENDING", nil
```

Comment:
```suggestion
			return result, "PENDING", nil
```
the first returned value can not be nil